src/third-party/imtui/imtui-impl-ncurses.cpp

```diff
@@ -153,13 +131,139 @@ void ImTui_ImplNcurses_Shutdown()
     // ref #11 : https://github.com/ggerganov/imtui/issues/11
     printf( "\033[?1003l\n" ); // Disable mouse movement events, as l = low
 
-    ImTui_ImplNCurses_Data* data = ImTui_ImplNCurses_GetBackendData();
+    ImTui_ImplNCurses_Data *data = ImTui_ImplNCurses_GetBackendData();
     if( data ) {
         delete data;
         ImGui::GetIO().BackendRendererUserData = nullptr;
     }
 }
 
+
+struct ImGuiKeyTranslation {
+    ImGuiKey key;
+    bool shift;
+    ImGuiKeyTranslation( ImGuiKey k, bool s = false ) :
+        key( k ),
+        shift( s )
+    {}
+};
+
+
+// returns ImGuiKey_None if it doesn't have an ImGui_Key counterpart
+// or if it is a character compatible with `ImGuiIO::addInputCharacter()`
+static ImGuiKeyTranslation ncurses_special_key_to_imgui_key( int key )
```

Comment:
This feels like it could/should be a map instead of a switch. I don't have a *strong* opinion on this, but I figure I'd point it out while I'm here.

Is there a reason for this is a #define? I'm not seeing usage in this PR that would require that.